src/main/java/net/imagej/ops/deconvolve/RichardsonLucyUpdate.java

```diff
@@ -52,44 +52,41 @@
  * @param <K>
  * @param <C>
  */
-@Plugin(type = Ops.Deconvolve.RichardsonLucy.class,
+@Plugin(type = Ops.Deconvolve.RichardsonLucyUpdate.class,
 	priority = Priority.HIGH_PRIORITY)
-public class RichardsonLucyImg<I extends RealType<I>, O extends RealType<O>, K extends RealType<K>, C extends ComplexType<C>>
-	extends AbstractFFTFilterImg<I, O, K, C>implements
-	Ops.Deconvolve.RichardsonLucy
+public class RichardsonLucyUpdate<T extends RealType<T>, I extends RandomAccessibleInterval<T>>
+	extends AbstractUnaryComputerOp<I, I> implements
+	Ops.Deconvolve.RichardsonLucyUpdate
 {
 
-	/**
-	 * max number of iterations
-	 */
-	@Parameter
-	private int maxIterations;
+	private BinaryHybridCF<I, I, I> mul = null;
 
-	/**
-	 * indicates whether to use non-circulant edge handling
-	 */
-	@Parameter(required = false)
-	private boolean nonCirculant = false;
+	@Override
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	public void initialize() {
 
-	/**
-	 * indicates whether to use acceleration
-	 */
-	@Parameter(required = false)
-	private boolean accelerate = false;
+		// TODO: comment in when problem with initialize is fixed
+		// mul = (BinaryHybridOp) Hybrids.binary(ops(),
+		// IIToIIOutputII.Multiply.class,
+		// RandomAccessibleInterval.class, RandomAccessibleInterval.class,
+		// RandomAccessibleInterval.class);
+	}
 
 	/**
-	 * run RichardsonLucyRAI
+	 * performs update step of the Richardson Lucy Algorithm
 	 */
 	@Override
-	public void runFilter(RandomAccessibleInterval<I> raiExtendedInput,
-		RandomAccessibleInterval<K> raiExtendedKernel, Img<C> fftImg,
-		Img<C> fftKernel, Img<O> output, Interval imgConvolutionInterval)
-	{
+	public void compute1(I correction, I estimate) {
+
+		// TODO: delte these lines when problem in initialization is fixed
+		if (mul == null) {
```

Comment:
When I wrote these ops, I assumed that `in1()` and `in2()` cannot be null, just to make the code simpler and make the matching safer (if those parameters are null the op has no way to do conforms). Usually there are ways to pass instances instead of class objects as inputs values for matching. 

For your case, will this work (in `initialize()`):
```
mul = RAIs.binaryHybrid(ops(), Ops.Math.Multiply.class, in1(), in2());
```
As long as the `in1()` and `in2()` are not null in this op, he mul would be matched to `IIToIIOutputII`. (Using `Ops.Math.Multiply.class` might be a better practice, unless there is a reason for explicit matching) 
A side note: the input and output should have `final` modifier. You can "right click->Source->clean up" to do that automatically, if you don't already know.